packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -264,11 +298,26 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
       prev: this.tip,
       id: this.id.cid,
     }
+    // Null check is necessary to avoid `undefined` value that can't be encoded with IPLD
+    if (metadata.shouldIndex != null) {
```

Comment:
don't we also need to check if metadata itself is null/undefined, like we do `replace` above?  Otherwise we could get an error trying to access the `shouldIndex` field of `undefined` right?

Reply:
Yes good catch, I fixed it